meteor/lib/collections/PieceInstances.ts

```diff
@@ -0,0 +1,58 @@
+import { TransformedCollection } from '../typings/meteor'
+import { registerCollection, literal } from '../lib'
+import { Meteor } from 'meteor/meteor'
+import { IBlueprintPieceInstance, Time, IBlueprintResolvedPieceInstance } from 'tv-automation-sofie-blueprints-integration'
+import { createMongoCollection } from './lib'
+import { Piece } from './Pieces'
+import { PartInstance } from './PartInstances'
+
+export interface PieceInstance extends IBlueprintPieceInstance {
+	/** Whether this PieceInstance is a temprorary wrapping of a Piece */
+	readonly isTemporary?: boolean
+
+	_id: string
+	/** The rundown this piece belongs to */
+	rundownId: string
+	/** The part instace this piece belongs to */
+	partInstanceId: string
+
+	piece: Piece
+}
+
+export interface ResolvedPieceInstance extends PieceInstance, IBlueprintResolvedPieceInstance {
+	piece: Piece
+}
+
+export function wrapPieceToTemporaryInstance (piece: Piece, partInstanceId: string): PieceInstance {
+	return literal<PieceInstance>({
+		isTemporary: true,
+		_id: `${piece._id}_tmp_instance`,
+		rundownId: piece.rundownId,
+		partInstanceId: partInstanceId,
+		piece: piece
+	})
+}
+
+export function findPieceInstanceOrWrapToTemporary (pieceInstances: PieceInstance[], partInstanceId: string, piece: Piece): PieceInstance {
+	return pieceInstances.find(instance => instance.piece._id === piece._id) || wrapPieceToTemporaryInstance(piece, partInstanceId)
```

Comment:
For now I am dropping this method. I don't think there should be any real need for it as we should be taking either 100% real or 100% temporary, otherwise the ui will be a confusing mess